hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/TableResolverImpl.java

```diff
@@ -135,25 +137,58 @@ public void createMapping(Mapping mapping, boolean replace, boolean ifNotExists)
     }
 
     private Mapping resolveMapping(Mapping mapping) {
-        String type = mapping.type();
         Map<String, String> options = mapping.options();
+        String name = mapping.name();
+        String type = mapping.type();
+        String dataLink = mapping.dataLink();
+        List<MappingField> resolvedFields;
+        SqlConnector connector;
+
+        if (type == null) {
+            connector = extractConnector(dataLink);
 
-        SqlConnector connector = connectorCache.forType(type);
-        List<MappingField> resolvedFields = connector.resolveAndValidateFields(
+        } else {
+            connector = connectorCache.forType(type);
+        }
+        resolvedFields = connector.resolveAndValidateFields(
                 nodeEngine,
                 options,
                 mapping.fields(),
                 mapping.externalName()
         );
+
         return new Mapping(
                 mapping.name(),
                 mapping.externalName(),
+                mapping.dataLink(),
                 type,
+                mapping.objectType(),
                 new ArrayList<>(resolvedFields),
                 new LinkedHashMap<>(options)
         );
     }
 
+    private SqlConnector extractConnector(String dataLink) {
+        SqlConnector connector;
+        assert dataLink != null;
```

Comment:
parameter should be `@Nonnull`. If this is invariant of `Mapping` it should be checked there (in constructor) instead of being scattered as asserts in many places.